be/src/pipeline/pipeline_x/operator.h

```diff
@@ -21,25 +21,25 @@
 
 namespace doris::pipeline {
 
-#define CREATE_LOCAL_STATE_RETURN_IF_ERROR(local_state)                                 \
-    auto _sptr = state->get_local_state(id());                                          \
-    if (!_sptr) return Status::InternalError("could not find local state id {}", id()); \
-    auto& local_state = _sptr->template cast<LocalState>();
-
-#define CREATE_SINK_LOCAL_STATE_RETURN_IF_ERROR(local_state)                            \
-    auto _sptr = state->get_sink_local_state(id());                                     \
-    if (!_sptr) return Status::InternalError("could not find local state id {}", id()); \
-    auto& local_state = _sptr->template cast<LocalState>();
-
-#define CREATE_LOCAL_STATE_RETURN_NULL_IF_ERROR(local_state) \
-    auto _sptr = state->get_local_state(id());               \
-    if (!_sptr) return nullptr;                              \
-    auto& local_state = _sptr->template cast<LocalState>();
-
-#define CREATE_SINK_LOCAL_STATE_RETURN_NULL_IF_ERROR(local_state) \
-    auto _sptr = state->get_sink_local_state(id());               \
-    if (!_sptr) return nullptr;                                   \
-    auto& local_state = _sptr->template cast<LocalState>();
+#define CREATE_LOCAL_STATE_RETURN_STATUS_IF_ERROR(local_state) \
```

Comment:
why use macro，it's better to use inline function not macro. inline function is good for debug and can get a clean namespace. macro is only for define scope or create an anonymous variable

Reply:
could  not agree more

Reply:
But this is used to create the local_state variable.